python/docs/cli.rst

```diff
@@ -0,0 +1,231 @@
+
+UI Commands
+===========
+
+Available Commands
+------------------
+
+.. code-block:: bash
+
+    Usage: nessie [OPTIONS] COMMAND [ARGS]...
+
+      Nessie cli tool.
+
+      Interact with Nessie branches and tables via the command line
+
+    Options:
+      --version
+      --help     Show this message and exit.
+
+    Commands:
+      branch       Branch operations.
+      cherry-pick  Transplant HASHES onto current branch.
+      config       Set and view config.
+      contents     Contents operations.
+      log          Show commit log.
+      merge        Merge BRANCH into current branch.
+      remote       Set and view remote endpoint.
+      tag          Tag operations.
+
+
+Config Command
+--------------
+Used to set config parameters found in ``default_config.yaml`` and to set the default context. To set default context use
+``nessie config --add ref main``, all operations that are ref specific will happen on that ref unless otherwise specified.
+
+.. code-block:: bash
+
+    Usage: nessie config [OPTIONS] [KEY]
+
+      Set and view config.
+
+    Options:
+      --get TEXT    get config parameter
+      --add TEXT    set config parameter
+      -l, --list    list config parameters
+      --unset TEXT  unset config parameter
+      --type TEXT   type to interpret config value to set or get. Allowed options:
+                    bool, int
+
+      -h, --help    Show this message and exit.
+
+
+Branch Command
+--------------
+Perform operations on branches: create, delete, modify and reassign.
+
+.. code-block:: bash
+
+    Usage: nessie branch [OPTIONS] [BRANCH] [NEW_BRANCH]
+
+      Branch operations.
+
+      BRANCH name of branch to list or create/assign NEW_BRANCH name of branch
+      to assign from or rename to
+
+      Examples:
+
+          nessie branch -l -> list all branches
+
+          nessie branch -l main -> list only main
+
+          nessie branch -d main -> delete main
+
+          nessie branch -> list all branches
+
+          nessie branch main -> create branch main at current head
+
+          nessie branch main test -> create branch main at head of test
+
+          nessie branch -f main test -> assign main to head of test
+
+    Options:
+      -l, --list     list branches
+      -d, --delete   delete a branch
+      -f, --force    force branch assignment
+      --json         write output in json format.
+      -v, --verbose  Verbose output.
+      -c, --condition TEXT  Conditional Hash. Only perform the action if branch
```

Comment:
I would assume this is required except in the cases where force is present. Is that correct?